### DISCUSSION TITLE
DwC sample data

### DISCUSSION THREAD

**Discussion by miekeGR** [ternaustralia/ontology_tern#53] [Modelling]
Please find attached 2 datasets that I have created from downloading data from ALA using their full Darwin Core download option. 
The Feaella from Pilbara dataset is a smaller, easier one with material samples as the feature of interest, that comes from a single institution (WA Museum). It has fairly wide coverage of standard DwC fields. 
The Onychophora (a fascinating group that uses slime to catch prey, including funnel web spiders!) around Perth dataset is a larger dataset that includes both material samples and human observations from a number of different sources. I was wondering how you might tackle it. Would you break it up per "dataresourceUid" identified in column A?
Can you let me know if these types of sample data are useful or if you had something else in mind to start with for the cookbook. Feel free to compartmentalise the data as much as you want for building the cookbook.

Thanks again so much for going through this mapping exercise with us. It's been so helpful to see how to do this properly.

Cheers,

Mieke

[Feaella from Pilbara records-2021-12-01.csv](https://github.com/ternaustralia/ontology_tern/files/7632473/Feaella.from.Pilbara.records-2021-12-01.csv
[Onychophora around Perth records-2021-11-30.csv](https://github.com/ternaustralia/ontology_tern/files/7632476/Onychophora.around.Perth.records-2021-11-30.csv)
)

**Comment by edmondchuc** [comment]
Hi @miekeGR, are you able to provide the source (link) where the data was downloaded from? I'd like to see if there is any other background information.

**Reply by miekeGR** [reply]
Hi @edmondchuc , 

My apologies, I definitely should have done that. 

**The _Feaella_ dataset has the following details:**

"The DOI for this download is https://doi.org/10.26197/ala.26fdc11f-107e-45fa-9aab-3aead9083137

When using this dataset please use the following citation:
Atlas of Living Australia occurrence download at https://biocache.ala.org.au/occurrences/search?q=lsid%3Aurn%3Alsid%3Abiodiversity.org.au%3Aafd.taxon%3A551bfe0b-dea1-4a03-b919-6f72e356e47d&qualityProfile=ALA&wkt=MULTIPOLYGON(((116.619873046875%20-20.718868625384793%2C116.619873046875%20-22.313237278525385%2C121.014404296875%20-22.313237278525385%2C121.014404296875%20-20.718868625384793%2C116.619873046875%20-20.718868625384793))) accessed on 01 December 2021."

**The Onychophora dataset has these details:** 

"The DOI for this download is https://doi.org/10.26197/ala.52faf6d5-7e55-4d1f-aa0c-792e87ee92b6

When using this dataset please use the following citation:
Atlas of Living Australia occurrence download at https://biocache.ala.org.au/occurrences/search?q=lsid%3Aurn%3Alsid%3Abiodiversity.org.au%3Aafd.taxon%3Af9293902-ab93-4293-b0d1-743d9b6cbfcf&qualityProfile=ALA&wkt=MULTIPOLYGON(((115.22186279296874%20-31.396725706951145%2C115.22186279296874%20-32.80199328284486%2C116.83685302734375%20-32.80199328284486%2C116.83685302734375%20-31.396725706951145%2C115.22186279296874%20-31.396725706951145))) accessed on 30 November 2021."

**Reply by edmondchuc** [reply]
Thanks for that @miekeGR. 

@dr-shorthair and I will have a look at these datasets and use them as a primary use case for mapping Darwin Core data to the TERN Ontology.

**Comment by dr-shorthair** [comment]
I've started by classifying the non-empty columns of the Feaella spreadsheet according to what I think are the relevant classes from the TERN Ontology. In many cases I've also been able to suggest an actual path match. The full mapping would also include some relationships between the several entities presented in a single row in the spreadsheet. 

I have NOT attempted to map the description of the taxon. 

There appears to be a gap in the TERN Ontology for the 'Record' - a kind of container for the sub-graph corresponding to a row in the spreadsheet.

**Reply by dr-shorthair** [reply]
Note that I have inferred a Site and SiteVisit from information provided. I know that these are new concepts, but they are important and useful in teasing out the real meaning of the DwC 'Occurrence' record. Apart from the work we've done designing the TERN Ontology, I have been involved in discussions in TDWG (the maintainers of DwC) where this more event-oriented model is becoming more clear as well.

**Reply by dr-shorthair** [reply]
[Feaella.from.Pilbara.records-2021-12-01.xlsx](https://github.com/ternaustralia/ontology_tern/files/7664221/Feaella.from.Pilbara.records-2021-12-01.xlsx)

**Reply by miekeGR** [reply]
Thanks @dr-shorthair . I have had a look at your mapping assignments and I agree, a 'Record' class would be very useful for describing this kind of data. I do have some follow-up questions with some of the other assignments that I will break down into individual comments for easier replies.

**Reply by edmondchuc** [reply]
@dr-shorthair I agree a Darwin Core `Record` is a kind of container that groups many of the related information in the row of the spreadsheet together. I think to capture the information on the `Record` we need to denormalise it across the concepts found in the TERN Ontology. In this way, a `Record` acts very similar to what an `ObservationCollection` provides for the normalisation of data across related `Observation` instances.

I am not sure if it is worth adding a `Record` class when we can denormalise the data. What do you both think?

**Reply by dr-shorthair** [reply]
In the example [attached below](https://github.com/ternaustralia/ontology_tern/files/7673124/feaella.ttl.txt) I 'created' a few new resources in a notional `bdr:` namespace to fill some holes. One of these is `bdr:Record`. I used it as a container whose members are the Site, Sampling, Sample and Observations, and also to hold some metadata which did not seem to belong anywhere else. It is really just a graph. We will need to understand if there is a real requirement that cannot be satisfied a better way. But at this stage I just wanted to show how the information in the spreadsheet could be captured.

**Reply by edmondchuc** [reply]
See https://github.com/ternaustralia/ontology_tern/discussions/53#discussioncomment-1802203 where a `Record` is modelled as a `tern:RDFDataset`.

**Comment by miekeGR** [comment]
Sites and Site Visit. For point observations, how precise is the requirement of the geometry to determine if one Site is the same as another Site i.e. you would use the same IRI between isolated datasets?

**Reply by dr-shorthair** [reply]
I can't answer that - it would be a business rule, not an ontology or encoding issue

**Reply by kheerand** [reply]
@dr-shorthair I agree at one level that this is a business decision.  Problem is that if we work towards a federated graph in the future, we run the risk of different definitions of what constitutes a site being made by different members of the federation.

Maybe this is a whacky idea but I wonder if the following would work around this:

1. every point observation gets an site IRI
2. predicates are added to denote that `SiteA owl:sameAs SiteB` based on business logic (included in this is also the prov of which member's business logic resulted in the addition of this link).

This would allow WA and DAWE to have slightly different definitions of what constitutes the _same_ site and both versions preserved in the data.

Or have I got something completely wrong in my thinking here?

**Reply by kheerand** [reply]
either way, it is clear that a business decision is needed around when two observations are considered as being on the same site.  I'm just starting off the governance group for the BDR and I will add this to the list of things they've got to decide upon.

@miekeGR, would you happen to have a few examples that can be used to illustrate this for the gov. group?

**Reply by miekeGR** [reply]
What you're proposing sounds good to me @kheerand .

In terms of examples, are you after examples from different datasets that may potentially look like the same Site? If so, I think I can help you out.

**Reply by edmondchuc** [reply]
I'd argue that sites and site visits are dependent on the survey protocol. Here are some survey protocols where "sites/plots" are well defined.

- AusPlots Rangelands - [AusPlots Rangelands Survey Protocols Manual v1.2.9](https://www.researchgate.net/profile/Ben_Sparrow/publication/266262164_AusPlots_Rangelands_Survey_Protocols_Manual_v_129/links/542ba4ad0cf29bbc126a8c40/AusPlots-Rangelands-Survey-Protocols-Manual-v-129.pdf)
- QLD CORVEG - [Methodology for surveying and mapping regional ecosystems and vegetation communities in Queensland v5](https://www.publications.qld.gov.au/dataset/d8244c14-d879-4a11-878c-2b6d4f01a932/resource/6dee78ab-c12c-4692-9842-b7257c2511e4/fs_download/methology-mapping-re-vegetation-v5.pdf)

In AusPlots Rangelands, plots (sites) are established and set up for repeated visits (multiple site visits). These AusPlots sites are also reused for other TERN purposes. Example, our TERN Ecosystem Processes team reuse AusPlots sites to perform other surveys/monitoring.  These other activities and artefacts link back to the AusPlots site with the same IRI. 

However, in CORVEG, sites are not set up permanently (at least currently). The CORVEG database establishes new sites even if the site is within the area of a previously surveyed site. This may change in the future since the new CORVEG database schema now supports the notion of site visits and I'm guessing they will record revisits as a new site visit instead of establishing a new site as they did before.

If by chance a CORVEG site is within an AusPlots site, we would still not consider these to be the same site. The reason is because you may record protocol-specific information about these sites, e.g., how it was set up, what was left behind, where are the permanent markers, etc. 

I don't know if these Darwin Core datasets have survey protocol manuals or not since it is aggregated data from ALA. It may be possible that the data were collected opportunistically (i.e., without the setup of a site). But I also don't see anything wrong with inferring the site and site visit information as mentioned by @dr-shorthair in https://github.com/ternaustralia/ontology_tern/issues/59#issuecomment-986473319, if it fits the mental model of the BDR.

As a business decision, for site-less data I’d advise against combining or inferring sites based on its location, unless it was specified in the survey protocol. It may skew the meaning of the data since biodiversity measures are scale and survey-method dependent.

I'd also caution the use of `owl:sameAs` as it may have unintended effects when querying with OWL inferencing enabled.

**Reply by miekeGR** [reply]
That's a great point @edmondchuc the protocol (and I guess the feature of interest also) would determine if the the sites could be seen as the same, not just the locality.

**Reply by miekeGR** [reply]
The bdr-ibsa-sample-data https://github.com/ternaustralia/bdr-ibsa-sample-data/ i.e. the first dataset we went through together is an interesting example. It contains 4 specimens: 3 are individuals of an operational taxonomic unit from the genus _Paradraculoides_ and the 4th is a specimen of _Paradraculoides anachoretus_. These four specimens were collected from a troglofauna trap on the same day, from the same bore hole (that has a nominated siteName in the dataset) of identical coordinates. In this case, I would say each observation was from a single site. 

In the second example we have gone through https://github.com/ternaustralia/bdr-faealla-worked-example I would like to draw your attention to the three specimens collected in 2011. They are all the same species (_Feaella (Tetrafeaella) tealei_), collected from a locality with the same coordinates; however, each was collected using a different method one by hand, one by wet pitfall trap, one using a Tullgren funnel. Should each of those specimens be from a different site or the same one? Interesting...

**Reply by edmondchuc** [reply]
> Should each of those specimens be from a different site or the same one?

I would say looking at the values in `locationRemarks`, the ones with the same values are the same site.

> They are all the same species (Feaella (Tetrafeaella) tealei), collected from a locality with the same coordinates; however, each was collected using a different method one by hand, one by wet pitfall trap, one using a Tullgren funnel

This would be expressed by the procedure in each of the sampling activities.

**Reply by dr-shorthair** [reply]
They are different acts-of-sampling (using different procedures),  having the same feature-of-interest (i.e. the site), generating different resultant samples (i.e. the individual specimens).

**Comment by miekeGR** [comment]
Attribute vs Parameter Type.

@dr-shorthair you assigned quite a few things to being Attributes including the fields dwc:individualCount and dwc:lifeStage. I did not see them in the tern:Attributes vocabulary, but I did find them in the tern:Parameter Type https://linkeddata.tern.org.au/viewer/tern/id/http://linked.data.gov.au/def/tern-cv/5699eca7-9ef0-47a6-bcfb-9306e0e2b85e 

Can you indicate how these vocabularies differ and how you would articulate that in the RDF?

**Reply by dr-shorthair** [reply]
I'll have to handball that to @edmondchuc - I understood that 'Attribute' was introduced to provide a slot for additional information that didn't have a native slot. So I just pushed some mappings into there, but had not (yet) considered the consequent requirements on controlled vocabularies. 

More important in what I did is to associate different ranges of a spreadsheet row with different classes or types from the ontology. This is the big improvement compared with the denormalised records you are getting from ALA etc.

**Comment by dr-shorthair** [comment]
I've bashed out an example RDF representation for the first two rows of the Feaella spreadsheet -
[feaella.ttl.txt](https://github.com/ternaustralia/ontology_tern/files/7673124/feaella.ttl.txt)

(I had to add .txt to the suffix for compatibility with GitHub.)

**Reply by dr-shorthair** [reply]
@miekeGR see if this makes sense to you

**Reply by miekeGR** [reply]
Hi @dr-shorthair , thanks for sharing that. I will look at it now.

**Comment by edmondchuc** [comment]
I've made an attempt at mapping the [Faealla dataset](https://doi.org/10.26197/ala.26fdc11f-107e-45fa-9aab-3aead9083137).

Here is the spreadsheet [mapped-records-2021-12-01.xlsx](https://github.com/ternaustralia/ontology_tern/files/7673814/mapped-records-2021-12-01.xlsx).

I'll follow up with an RDF view of the mapped data.

We can compare our interpretations and have a discussion. The final results will be documented in the cookbook.

**Reply by miekeGR** [reply]
Hi @edmondchuc I have made a start on diagramming your data mapping using lucidchart. I am sure there is a more efficient way to get this right, but I chose this method to explore my understanding of the mapping and to get a feel of where I think some gaps may lie that you might be able to help with.
I have done some things a little differently. Note, I didn't put in all the relationship names (not easy with space) and where I haven't connected the dataset class in lines, I've written it out as a property. I acknowledge, my made up IRIs are not ideal, but just a shortcut to get things on the page. 
With internal discussion at Gaia, we were thinking that we need to group the taxon identification observations together. So I have put them in as an Observation Collection.
Purple: Work-in-progress
Orange: Information not present in the dataset (inferred through search)
Red: Should be IRIs (either people, institutions, publications or new vocab)
I have flagged the “habitat” in red because it is in the “feature type” vocabulary and I was wondering if you can use it as an observed property even though it is not in the “parameters” vocabulary?

New vocabs that could be useful: type-status, lifestage, license
Comment: IRI for publications? - would be good, but might be difficult to populate from legacy data.
I haven’t put in the dwc:countryCode attribute. I wasn’t sure how to relate it to the site.

![Feaella test 20211209 (4)](https://user-images.githubusercontent.com/80197455/146051125-185ed4f2-74f9-4e82-b13e-8554b9e6aa23.png)

**Reply by edmondchuc** [reply]
Here's the repository of the code to generate the output data. https://github.com/ternaustralia/bdr-faealla-worked-example. 

It's changed a bit from the mapping spreadsheet uploaded above. The new mapping spreadsheet link is [here](https://docs.google.com/spreadsheets/d/1p3scX7z6wPQ0vtG-Bo_yoYcvRRs8muGm/edit?usp=sharing&ouid=108129827562056706312&rtpof=true&sd=true).

See [output-row-2.ttl](https://github.com/ternaustralia/bdr-faealla-worked-example/blob/ffcb897a8374c0eef7f79641706e990a5c0e74d5/output-row-2.ttl). It is the output of a single row as RDF Turtle. 

- There are two samplings, one for the **occurrence** and the other for the **specimen**. 
- There are two observations made on the occurrence sample for **life stage** and **sex**.
- There are a few observations on the **specimen** which I'm yet to add to the RDF output. I will add it in once I'm back from leave in the new year. The missing observations are marked in yellow in column A in the [spreadsheet](https://docs.google.com/spreadsheets/d/1p3scX7z6wPQ0vtG-Bo_yoYcvRRs8muGm/edit?usp=sharing&ouid=108129827562056706312&rtpof=true&sd=true).
- A **record's** information is represented with `tern:RDFDataset`. 

From the original mapping in the opening post to the new [mapping](https://docs.google.com/spreadsheets/d/1p3scX7z6wPQ0vtG-Bo_yoYcvRRs8muGm/edit?usp=sharing&ouid=108129827562056706312&rtpof=true&sd=true), I've removed sites and site visits. I believe these occurrence records are likely opportunistic. See related discussion on how sites and site visits are usually determined by the survey protocol in https://github.com/ternaustralia/ontology_tern/discussions/53#discussioncomment-1776326.

@miekeGR I will have a look at your diagram in more detail once I'm back from leave in the new year.

**Reply by miekeGR** [reply]
Many thanks @edmondchuc . I am really looking forward to looking at your work. 

Don't rush on the review on my diagram. I've realised a few errors and made a couple of changes. Will upload a revised copy later. 

Enjoy your break and see you next year :-D

**Reply by edmondchuc** [reply]
@miekeGR I've updated the repository https://github.com/ternaustralia/bdr-faealla-worked-example with observations on the specimen.

There are two observations on a specimen:
- **status type** 
- **taxonomic information**

See the worked example RDF output.

- RDF Turtle snippet of a single row: [output-row-2.ttl](https://github.com/ternaustralia/bdr-faealla-worked-example/blob/ffcb897a8374c0eef7f79641706e990a5c0e74d5/output-row-2.ttl)
- RDF Turtle snippet of the complete CSV: [output.ttl](https://github.com/ternaustralia/bdr-faealla-worked-example/blob/ffcb897a8374c0eef7f79641706e990a5c0e74d5/output.ttl)

A description of the modelling assumptions are also recorded in the [README.md](https://github.com/ternaustralia/bdr-faealla-worked-example/blob/master/README.md) of the repository.

**Reply by dr-shorthair** [reply]
Many charting mechanisms can help come to terms with the models. 
But they won't capture the labelled-directed-graph unless you also label the connectors. 

Also note that while some special relationships including 'is-subclass-of', 'is-disjoint-with' are between classes, all the ones with domain semantics (has-feature-of-interest, has-result, used-procedure, site-visited, is-sample-of ...) only occur between individual members of classes
